### PR TITLE
Move collision detection and movements to server side

### PR DESCRIPTION
**Pull request by codymaki**
## Changes

- moved collision detection to server side
- refactored enemy state to be at enemy level to allow for multiple enemies
- updated socket logic to receive asset location data
- removed hitbox display logic from front end, that will need to be refactored to account for new backend code.
- moved animation logic to backend
- refactored player logic and moved physics to backend, renaming from ninja

## References

- https://github.com/codymaki/Bokudos-server/pull/2 - server side code

## Screenshots

**example with two players and a bunch of enemies**
image

### CODE REVIEW

File: src/debug/debugMode.ts [hunk 1]
```diff
@@ -150,17 +150,17 @@ export class DebugMode {
...
-      State.gameState.canvas.ctx.strokeStyle = State.debugState.menuOptions[MenuOptions.ENEMY_HITBOX].color;
-      State.gameState.canvas.ctx.lineWidth = State.debugState.menuOptions[MenuOptions.ENEMY_HITBOX].lineWidth;
-
-      const box = State.enemyState.hitbox;
-      const { x, y } = RenderingUtilities.toScreenCoordinates(box.position);
-      const { w, h } = RenderingUtilities.toScreenDimensions(box.dimensions);
-      State.gameState.canvas.ctx.strokeRect(x, y, w, h);
-    }
-  }
+  // static drawEnemyHitbox() {
```
Comment (BadassBison) [inline src/debug/debugMode.ts:153]: Why is this commented out?

File: src/objects/enemy.ts [hunk 2]
```diff
@@ -1,40 +1,40 @@
...
-                y: this.state.position.y
-            },
-            dimensions: {
-                w: this.state.currentImage.width / this.state.SPRITE_SIZER,
-                h: this.state.currentImage.height / this.state.SPRITE_SIZER
-            }
-        };
     }
 
     updateProperties(keys: Keys): void {
```
Comment (BadassBison) [inline src/objects/enemy.ts:37]: Do we need this method anymore?

File: src/states/gameState.ts [hunk 3]
```diff
@@ -9,6 +9,7 @@ import { PhysicsEngine } from '../engines/physicsEngine';
 export class GameState {
     canvas: CanvasElement;
     assets: UpdateObject[];
+    assetMap: Map<string, UpdateObject>;
```
Comment (BadassBison) [inline src/states/gameState.ts:12]: Can we use this opposed to the array?

**Review by BadassBison — APPROVED** [review]

File: src/debug/debugMode.ts (hunk 1 above)
Comment (codymaki) [reply src/debug/debugMode.ts:153]: A lot of the hitbox debug stuff is going to be broken for a bit while I move things to the server side. I am deleting or commenting out things as needed.

File: src/objects/enemy.ts (hunk 2 above)
Comment (codymaki) [reply src/objects/enemy.ts:37]: No I will be removing this with the next big push. Along with some other refactor/cleanup as stuff moves to server side.

File: src/states/gameState.ts (hunk 3 above)
Comment (codymaki) [reply src/states/gameState.ts:12]: Yeah this map will completely replace the assets array, I am just leaving both while I test and move things over to server.

**Comment by codymaki** [comment]
@justinjordan fyi, I think I broke the audio part here. I can take a look at adding it back. I think the only thing we need to do is add back the sound stuff you had added to ninja.ts but add that where necessary to player.ts. Though it will be slightly different since before it was just doing the attack stuff on the front end but now that is doing done in the back end. I added an item in Trello so we don't forget about this. https://trello.com/c/znAwkDL0/41-fix-sound-logic-that-was-broken-from-moving-physics-to-backend

**Pull request merged**